Fix clippy lint

tottoto opened the pull request:
Fixes clippy lints.

LucioFranco reviewed (commented):
LGTM just want to see the eq stuff backed out into codegen and have one qusetion

tottoto commented on tonic/src/transport/service/io.rs:
```diff
@@ -70,7 +70,7 @@ impl AsyncWrite for BoxedIo {
     Io(IO),
     #[cfg(feature = "tls")]
-    TlsIo(TlsStream<IO>),
+    TlsIo(Box<TlsStream<IO>>),
```
This fixes [large_enum_variant](https://rust-lang.github.io/rust-clippy/master/#large_enum_variant).

tottoto commented on tonic-build/src/prost.rs:
```diff
@@ -15,6 +15,11 @@ use std::{
 ///
 /// Use [`compile_protos`] instead if you don't need to tweak anything.
 pub fn configure() -> Builder {
+    let type_attributes = vec![(
+        ".".to_string(),
+        "#[allow(clippy::derive_partial_eq_without_eq)]".to_string(),
+    )];
```
As adding attributes might be only be configured via prost api, it seems like we need to set this as the default. However I'm not sure that this is an appropriate approach.

LucioFranco replied:
Ugh I wonder if we should just fix this in prost then? Somewhere in this macro impl https://github.com/tokio-rs/prost/blob/master/prost-derive/src/lib.rs#L21 since really the problem is I don't want code generated stuff to impl Eq or we can put this in prost-build?

tottoto replied:
I published https://github.com/tokio-rs/prost/pull/775 as to try solving this.

tottoto commented:
I have to remove clippy attributes from each fille which includes proto, and update generated codes.

LucioFranco commented:
The ci failure is interesting I think maybe you didn't run it with the clean up feature enabled?

LucioFranco commented:
Working on that now https://github.com/tokio-rs/prost/pull/777

LucioFranco commented:
ok its been published as v0.11.4

tottoto commented:
I updated prost-build to 0.11.4 in https://github.com/hyperium/tonic/pull/1181/commits/e7efd338322b97addc95d1b7d9d4330a17a38837 and fixed some warnings in https://github.com/hyperium/tonic/pull/1181/commits/7c455f70a731d23a5f2063bb4e228859ef183b4f.

tottoto commented:
I reverted https://github.com/hyperium/tonic/pull/1181/commits/7fe1d32111c76bd51bc7ce6bd0657d481c27d9a2 in https://github.com/hyperium/tonic/pull/1181/commits/728b040d584fcbb08a31d2075c91f072ff4953b5 as it was took over by prost-build 0.11.4.

LucioFranco commented on tests/root-crate-path/src/main.rs:
```diff
@@ -1,4 +1,4 @@
-#[derive(Clone, PartialEq, ::prost::Message)]
+#[derive(Clone, PartialEq, Eq, ::prost::Message)]
```
This should get an allow with the new version of prost right?

tottoto replied:
This looks like my mistake. I'll fix this.

tottoto replied:
Fixed.

LucioFranco reviewed (approved):
Thanks!
